src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseIntegrationTestService.java

```diff
@@ -276,6 +279,10 @@ void tearDown() throws IOException {
         if (remoteRepo2File != null && remoteRepo2File.exists()) {
             FileUtils.deleteDirectory(remoteRepo2File);
         }
+        var repoDownloadCloneDir = new File(repoDownloadClonePath);
+        if (repoDownloadCloneDir.exists()) {
+            FileUtils.deleteDirectory(repoDownloadCloneDir);
+        }
```

Comment:
This will not work. You now try to delete all repositories in the folder, but due to parallel test execution, this could lead to deleting a repository that is still needed by a different test.

Reply:
Good point. I changed it to use temp files